berserker-mqtt/src/main/java/io/smartcat/berserker/mqtt/worker/AsyncMqttWorker.java

```diff
@@ -0,0 +1,77 @@
+package io.smartcat.berserker.mqtt.worker;
+
+import java.util.Map;
+
+import org.eclipse.paho.client.mqttv3.IMqttToken;
+import org.eclipse.paho.client.mqttv3.MqttAsyncClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+
+import io.smartcat.berserker.api.Worker;
+
+/**
+ * MQTT worker which publishes messages in an asynchronous fashion.
+ */
+public class AsyncMqttWorker implements Worker<Map<String, Object>>, AutoCloseable {
+
+    private static final String TOPIC = "topic";
+    private static final String QOS = "qos";
+    private static final String PAYLOAD = "payload";
+
+    private final MqttAsyncClient client;
+    private final MqttConnectOptions connectOptions;
+    private volatile boolean connected = false;
+
+    /**
+     * Constructs MQTT asynchronous worker with specified <code>client</code> and <code>connectOptions</code> to be
+     * used.
+     *
+     * @param client Asynchronous MQTT client which will publish messages.
+     * @param connectOptions Options to use to connect to MQTT broker.
+     */
+    public AsyncMqttWorker(MqttAsyncClient client, MqttConnectOptions connectOptions) {
+        this.client = client;
+        this.connectOptions = connectOptions;
+    }
+
+    @Override
+    public void accept(Map<String, Object> message) {
+        ensureConnected();
+        String topic = (String) message.get(TOPIC);
+        int qos = (Integer) message.get(QOS);
+        String payload = (String) message.get(PAYLOAD);
+        MqttMessage mqttMessage = new MqttMessage(payload.getBytes());
+        mqttMessage.setQos(qos);
+        try {
+            client.publish(topic, mqttMessage);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
```

Comment:
Abstract these two methods since its the same code or initialize the same worker with different MqttClient reference.

Reply:
Tried it, `MqttClient` and `AsyncMqttClient` are not part of the same class hierarchy. Only to create our wrapper around it and pass that.